saleor/order/tasks.py

```diff
@@ -21,3 +39,75 @@ def send_order_updated(order_ids):
     manager = get_plugins_manager()
     for order in Order.objects.filter(id__in=order_ids):
         manager.order_updated(order)
+
+
+def _batch_ids(iterable, batch_size=1):
+    length = len(iterable)
+    for index in range(0, length, batch_size):
+        yield iterable[index : min(index + batch_size, length)]
+
+
+def _queryset_in_batches(queryset):
+    ids = queryset.values_list("id", flat=True)
+    for ids_batch in _batch_ids(ids, ORDER_BATCH_SIZE):
+        yield ids_batch
+
+
+def _bulk_release_voucher_usage(order_ids):
+    voucher_orders = Order.objects.filter(
+        voucher=OuterRef("pk"),
+        id__in=order_ids,
+    )
+    Voucher.objects.filter(
+        Exists(voucher_orders),
+        usage_limit__isnull=False,
+    ).update(used=F("used") - 1)
+
+    voucher_customer_orders = Order.objects.filter(
+        voucher=OuterRef("voucher__id"),
+        user_email=OuterRef("customer_email"),
+        id__in=order_ids,
+    )
+    VoucherCustomer.objects.filter(Exists(voucher_customer_orders)).delete()
+
+
+def _call_expired_order_events(order_ids, manager):
+    orders = Order.objects.filter(id__in=order_ids)
+    for order in orders:
+        call_event(manager.order_expired, order)
+        call_event(manager.order_updated, order)
+
+
+def _expire_orders(manager, now):
+    qs = Order.objects.filter(
+        ~Exists(TransactionItem.objects.filter(order=OuterRef("pk"))),
+        ~Exists(Payment.objects.filter(order=OuterRef("pk"))),
+        status=OrderStatus.UNCONFIRMED,
+        channel__expire_orders_after__isnull=False,
+        channel__expire_orders_after__gt=0,
+        channel__expire_orders_after__lte=Extract(now - F("created_at"), "epoch"),
+        total_charged_amount=Decimal(0),
```

Comment:
`total_charged_amount=Decimal(0),` I think this is not needed. If you don't have any payments and any TransactionItem, then we don't have anything that could be charged.